src/shared/bg_pmove.cpp

```diff
@@ -4847,17 +4847,20 @@ void PmoveSingle( pmove_t *pmove )
 
 	// determine the time
 	pml.msec = pmove->cmd.serverTime - pm->ps->commandTime;
+	pm->ps->commandTime = pmove->cmd.serverTime;
 
-	if ( pml.msec < 1 )
-	{
-		pml.msec = 1;
-	}
-	else if ( pml.msec > 200 )
+	if ( pm->ps->pm_type == PM_SPECTATOR || pm->ps->pm_type == PM_NOCLIP )
 	{
-		pml.msec = 200;
+		// HACK: ignore the effects of timescale for spectators
```

Comment:
Instead of this, find where cg_flyspeed is plugged in and do the scaling there

Reply:
Well, I did that at first. Notice that this is in bg, so that you need to handle the custom code with `#ifdef` directives, but since you can't access cvars so you can't even do that easily. And for a smooth experience you should also find a way to compensate the other params that influence moving, like acceleration and friction. So the whole ends up being a much bigger mess.

Reply:
It's not related to BG. You can do it like this:
```
--- a/src/sgame/sg_active.cpp
+++ b/src/sgame/sg_active.cpp
@@ -632,7 +632,7 @@ void SpectatorThink( gentity_t *ent, usercmd_t *ucmd )
                        client->ps.pm_flags |= PMF_QUEUED;
                }

-               client->ps.speed = client->pers.flySpeed;
+               client->ps.speed = client->pers.flySpeed / atof(Cvar::GetValue("timescale").c_str());
                client->ps.stats[ STAT_STAMINA ] = 0;
                client->ps.stats[ STAT_FUEL ] = 0;
                client->ps.stats[ STAT_MISC ] = 0;
@@ -1994,7 +1994,7 @@ void ClientThink_real( gentity_t *self )
        // set speed
        if ( client->ps.pm_type == PM_NOCLIP )
        {
-               client->ps.speed = client->pers.flySpeed;
+               client->ps.speed = client->pers.flySpeed / atof(Cvar::GetValue("timescale").c_str());
        }
        else
        {
```
Yeah the friction is different. Who cares? BUT, I would not accept this patch in the current form, since it would trigger a lot of VM->engine syscalls to get the value.

We really need a cvar interface that lets a VM observe a cvar owned by the engine. Currently the VM can only subscribe to updates on cvars created by it.

Reply:
Well, the version I'm suggesting has no such "syscall" problem :)

Reply:
I'll give a try at the code tomorrow to see if that works. If yes, then LGTM

Reply:
That was indeed a veeeeeery long night. I checked, and it works, so as I said, I'm fine with this.